tests/unittests/ProvisionerTest.cpp

```diff
@@ -79,8 +79,8 @@ TEST_F(ProvisionerTest, provisionDag) {
   }
 
   CompilationContext cctx;
-  auto provisioner = Provisioner(devices);
```

Comment:
curious why this change

Reply:
With the added mutex the default copy constructor is deleted, but a heap allocation isn't needed, changed back to stack allocation.